packages/block-library/src/block/edit.js

```diff
@@ -6,207 +6,182 @@ import { partial } from 'lodash';
 /**
  * WordPress dependencies
  */
-import { Component } from '@wordpress/element';
-import { Placeholder, Spinner, Disabled } from '@wordpress/components';
-import { withSelect, withDispatch } from '@wordpress/data';
-import { __ } from '@wordpress/i18n';
 import {
 	BlockEditorProvider,
 	BlockList,
 	WritingFlow,
 } from '@wordpress/block-editor';
-import { compose } from '@wordpress/compose';
-import { parse, serialize } from '@wordpress/blocks';
+import { serialize } from '@wordpress/blocks';
+import { Placeholder, Spinner, Disabled } from '@wordpress/components';
+import { useDispatch, useSelect } from '@wordpress/data';
+import { useEffect, useState } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
 
 /**
  * Internal dependencies
  */
 import ReusableBlockEditPanel from './edit-panel';
 
-class ReusableBlockEdit extends Component {
-	constructor( { reusableBlock } ) {
-		super( ...arguments );
-
-		this.startEditing = this.startEditing.bind( this );
-		this.stopEditing = this.stopEditing.bind( this );
-		this.setBlocks = this.setBlocks.bind( this );
-		this.setTitle = this.setTitle.bind( this );
-		this.save = this.save.bind( this );
-
-		if ( reusableBlock ) {
-			// Start in edit mode when we're working with a newly created reusable block
-			this.state = {
-				isEditing: reusableBlock.isTemporary,
-				title: reusableBlock.title,
-				blocks: parse( reusableBlock.content ),
```

Comment:
To the recent issues you're experiencing, I observe the difference between the branch and master being that, at the initialization of state for a new reusable block, we'd previously parse the content of the temporary block. In the new implementation, the blocks state will be initialized to `null` instead, and never be updated.

I think we need to ensure the state is initialized with the parsed result, and optionally have effects in place which keep that state up to date in response to changes (is there an equivalent to what existed previously in `componentDidUpdate`?).

Per the prior concern about performance, we'd still want to make sure that the content is parsed only once for the initial state.

See: https://reactjs.org/docs/hooks-reference.html#lazy-initial-state

Reply:
Thanks. Something I don't understand is how detecting the existence of `reusableBlock` indicates a _newly created_ reusable block. I guess I don't quite understand what that if statement means.

The equivalent to the `componentDidUpdate` code were [these effects](https://github.com/WordPress/gutenberg/pull/21427/commits/78767e8362e37e4f6f44872346ece6f2257eecec#diff-ae0ba49328ef77ebd8fc08a86fee257aR92-R102). But since you wanted to stop using local state when not necessary, I removed those.

Reply:
Hm, I think it may have been a pretty fragile assumption based on how an _existing_ reusable block would need to be fetched (see `fetchReusableBlock` in prior `componentDidMount`) and thus there was an assumption that `reusableBlock` would only be available initially in the case of a newly converted reusable block. It seems like something that would be more reliable to check for `reusableBlock?.isTemporary`.

Reply:
Alright, I'm pushing a commit right now that might fix things... I'll let you know how it turns out. Thank you very much for all of your help so far. :smile:

Reply:
Well, it didn't work. I think I probably missed something, though. This is all rather difficult to wrap my head around. :confounded:

Reply:
Alright, I messed around with the code some more, and now I've got things working again! :partying_face:

There's one minor issue I've noticed, however: whenever you switch between editing/not-editing a reusable block, it momentarily appears empty (IOW, it temporarily reverts to just showing the empty Paragraph placeholder). It makes for some annoying jumpiness that definitely isn't ideal.

Reply:
It turns out that the momentary emptiness was easily fixed by switching from `useEffect` to `useLayoutEffect`. :smile: